tests/pausing_test.go

```diff
@@ -0,0 +1,436 @@
+				Expect(command()).To(Succeed())
+				tests.WaitForVMCondition(virtClient, vm, v1.VirtualMachinePaused, 30)
+
+				command = tests.NewRepeatableVirtctlCommand("pause", "vmi", "--namespace", tests.NamespaceTestDefault, vm.Name)
+				err := command()
+				Expect(err.Error()).To(ContainSubstring("VMI is already paused"))
+			})
+
+			It("[test_id:3060]should signal unpaused state with removed condition", func() {
+				runVM()
+				command := tests.NewRepeatableVirtctlCommand("pause", "vm", "--namespace", tests.NamespaceTestDefault, vm.Name)
+				Expect(command()).To(Succeed())
+				tests.WaitForVMCondition(virtClient, vm, v1.VirtualMachinePaused, 30)
+
+				command = tests.NewRepeatableVirtctlCommand("unpause", "vm", "--namespace", tests.NamespaceTestDefault, vm.Name)
+				Expect(command()).To(Succeed())
+				tests.WaitForVMConditionRemovedOrFalse(virtClient, vm, v1.VirtualMachinePaused, 30)
+			})
+
+			It("[test_id:3082]should gracefully handle unpausing again", func() {
+				runVM()
+				command := tests.NewRepeatableVirtctlCommand("pause", "vm", "--namespace", tests.NamespaceTestDefault, vm.Name)
+				Expect(command()).To(Succeed())
+				tests.WaitForVMCondition(virtClient, vm, v1.VirtualMachinePaused, 30)
+
+				command = tests.NewRepeatableVirtctlCommand("unpause", "vm", "--namespace", tests.NamespaceTestDefault, vm.Name)
+				Expect(command()).To(Succeed())
+				tests.WaitForVMConditionRemovedOrFalse(virtClient, vm, v1.VirtualMachinePaused, 30)
+
+				command = tests.NewRepeatableVirtctlCommand("unpause", "vm", "--namespace", tests.NamespaceTestDefault, vm.Name)
+				err := command()
+				Expect(err.Error()).To(ContainSubstring("VMI is not paused"))
+			})
+
+			It("[test_id:3085]should be stopped successfully", func() {
+
+				runVM()
+
+				By("Pausing the VM")
+				command := tests.NewRepeatableVirtctlCommand("pause", "vm", "--namespace", tests.NamespaceTestDefault, vm.Name)
+				Expect(command()).To(Succeed())
+				tests.WaitForVMCondition(virtClient, vm, v1.VirtualMachinePaused, 30)
+
+				By("Stopping the VM")
+				command = tests.NewRepeatableVirtctlCommand("stop", "--namespace", tests.NamespaceTestDefault, vm.Name)
+				Expect(command()).To(Succeed())
+
+				By("Checking deletion of VMI")
+				Eventually(func() bool {
+					_, err = virtClient.VirtualMachineInstance(vm.Namespace).Get(vm.Name, &v12.GetOptions{})
+					if errors.IsNotFound(err) {
+						return true
+					}
+					Expect(err).ToNot(HaveOccurred())
+					return false
+				}, 300*time.Second, 1*time.Second).Should(BeTrue(), "The VMI did not disappear")
+
+				By("Checking status of VM")
+				Eventually(func() bool {
+					vm, err := virtClient.VirtualMachine(vm.Namespace).Get(vm.Name, &v12.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					return vm.Status.Ready
+				}, 300*time.Second, 1*time.Second).Should(BeFalse())
+
+			})
+
+			It("should gracefully handle being started again", func() {
```

Comment:
I think we only started the VM once in this testcase, we need to start the VM here once again.

Reply:
The goal is to test what happens when you start a paused VM. And not to start it twice...?

Reply:
Ah! ok then.